src/libraries/Common/src/System/Data/Common/MultipartIdentifier.cs

```diff
@@ -68,7 +68,7 @@ private static bool IsWhitespace(char ch)
                 throw ADP.InvalidMultipartNameToManyParts(property, name, limit);
             }
 
-            if (-1 != leftQuote.IndexOf(separator) || -1 != rightQuote.IndexOf(separator) || leftQuote.Length != rightQuote.Length)
+            if (leftQuote.IndexOf(separator) != -1 || rightQuote.IndexOf(separator) != -1 || leftQuote.Length != rightQuote.Length)
```

Comment:
.... should we have a `NOT_FOUND` constant for the `IndexOf` comparisons?

Also, the current comparisons make my brain hurt for what they're saying ("not not found").  Should some of these be switched to `Contains`, and others to testing against a positive index?  (In another PR, obviously)